src/server/scripts/Northrend/Ulduar/Ulduar/boss_assembly_of_iron.cpp

```diff
@@ -15,751 +15,879 @@
-SDCategory: Ulduar - Ulduar
-EndScriptData */
-
 #include "ScriptMgr.h"
+#include "AreaBoundary.h"
 #include "InstanceScript.h"
 #include "MotionMaster.h"
 #include "ObjectAccessor.h"
 #include "ScriptedCreature.h"
 #include "SpellAuras.h"
 #include "SpellMgr.h"
 #include "SpellScript.h"
+#include "Spell.h"
 #include "TemporarySummon.h"
 #include "ulduar.h"
 
-enum AssemblySpells
+enum Spells
 {
     // General
-    SPELL_SUPERCHARGE                            = 61920,
-    SPELL_BERSERK                                = 47008, // Hard enrage, don't know the correct ID.
-    SPELL_KILL_CREDIT                            = 65195, // spell_dbc
+    SPELL_SUPERCHARGE                           = 61920,
+    SPELL_BERSERK                               = 62535,
+    SPELL_KILL_CREDIT                           = 65195, // Serverside spell
 
     // Steelbreaker
-    SPELL_HIGH_VOLTAGE                           = 61890,
-    SPELL_FUSION_PUNCH                           = 61903,
-    SPELL_STATIC_DISRUPTION                      = 44008,
-    SPELL_OVERWHELMING_POWER                     = 64637,
-    SPELL_ELECTRICAL_CHARGE                      = 61902,
+    SPELL_HIGH_VOLTAGE                          = 61890,
+    SPELL_STATIC_DISRUPTION                     = 44008,
+    SPELL_ELECTRICAL_CHARGE                     = 61902,
 
     // Runemaster Molgeim
-    SPELL_SHIELD_OF_RUNES                        = 62274,
-    SPELL_SHIELD_OF_RUNES_BUFF                   = 62277,
-    SPELL_SUMMON_RUNE_OF_POWER                   = 63513,
-    SPELL_RUNE_OF_DEATH                          = 62269,
-    SPELL_RUNE_OF_SUMMONING                      = 62273, // This is the spell that summons the rune
-    SPELL_RUNE_OF_SUMMONING_SUMMON               = 62020, // Spell that summons
+    SPELL_RUNE_OF_POWER_PRE_FIGHT               = 61975,
+    SPELL_RUNE_OF_POWER                         = 61973,
+    SPELL_SHIELD_OF_RUNES                       = 62274,
+    SPELL_SHIELD_OF_RUNES_BUFF                  = 62277,
+    SPELL_RUNE_OF_DEATH                         = 62269,
+    SPELL_RUNE_OF_SUMMONING                     = 62273,
+    SPELL_RUNE_OF_SUMMONING_SUMMON              = 62020,
 
     // Stormcaller Brundir
-    SPELL_CHAIN_LIGHTNING                        = 61879,
-    SPELL_OVERLOAD                               = 61869,
-    SPELL_LIGHTNING_WHIRL                        = 61915,
-    SPELL_LIGHTNING_TENDRILS                     = 61887,
-    SPELL_LIGHTNING_TENDRILS_VISUAL              = 61883,
-    SPELL_STORMSHIELD                            = 64187
+    SPELL_LIGHTNING_CHANNEL_PRE_FIGHT_VISUAL    = 61942,
+    SPELL_CHAIN_LIGHTNING                       = 61879,
+    SPELL_LIGHTNING_WHIRL                       = 61915,
+    SPELL_LIGHTNING_TENDRILS_VISUAL             = 61883,
+    SPELL_STORMSHIELD                           = 64187,
+    SPELL_RANDOM_AGGRO_PERIODIC                 = 61906
```

Comment:
This spell does not belong, it's used by Feral Defenders and another different mob, not by Brundir

Reply:
SPELL_RANDOM_AGGRO_PERIODIC                 = 61906

Reply:
It is. It's literally in the sniffs

Reply:
I managed to find a sniff and didn't see Brundir listed as the caster or the Aura holder anywhere, only 34288(Salvagebot Sawblade) and 34035(Feral Defender).

Reply:
Same

Reply:
```
ServerToClient: SMSG_AURA_UPDATE (0x2C22) Length: 57 ConnIdx: 1 Time: 09/29/2020 16:32:24.750 Number: 36878
UpdateAll: False
AurasCount: 1
[0] Slot: 5
[0] HasAura: True
[0] CastID: Full: 0xBC3CC84B603C74830077B50000F3457B Cast/3 R3890/S30645 Map: 603 (Ulduar) Low: 15943035
[0] SpellID: 61906 (Random Aggro Periodic (5 sec))
[0] SpellXSpellVisualID: 0
[0] Flags: 3 (NoCaster, Positive)
[0] ActiveFlags: 1
[0] CastLevel: 80
[0] Applications: 0
[0] ContentTuningID: 440
[0] HasCastUnit: False
[0] HasDuration: False
[0] HasRemaining: False
[0] HasTimeMod: False
[0] PointsCount: 0
[0] EstimatedPoints: 0
[0] HasContentTuning: False
UnitGUID: Full: 0x203CC84B602016400077B500007343CA Creature/0 R3890/S30645 Map: 603 (Ulduar) Entry: 32857 (Stormcaller Brundir) Low: 7553994
```

Reply:
Fair enough

Reply:
@Ovahlord in case that skill should be used by assembly of iron, it needs to be checked if auriaya should also use it, because they currently share script

https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/Ulduar/Ulduar/boss_auriaya.cpp#L579-L616
It does not seem very correct to use feral pounce (64478) when flying with lightning trendils 😆 

maybe both spellscript need to be corrected a little better